compiler/src/iree/compiler/Dialect/HAL/IR/HALOps.cpp

```diff
@@ -1508,17 +1531,156 @@ LogicalResult ExecutableOp::verify() {
 // hal.executable.export
 //===----------------------------------------------------------------------===//
 
+// Verifies that the export condition region matches the expected
+// signature. Returns success if the region is empty.
+static LogicalResult verifyExportConditionRegion(Operation *op,
+                                                 Region &region) {
+  if (region.empty())
+    return success();
+
+  // Verify one of the supported signatures.
+  bool validArguments = true;
+  if (region.getNumArguments() == 0) {
+    // Need at least a !hal.device.
+    validArguments = false;
+  } else if (!llvm::isa<IREE::HAL::DeviceType>(
+                 region.getArgument(0).getType())) {
+    // !hal.device must come first.
+    validArguments = false;
+  } else {
+    // All remaining arguments need to be of type index (today).
+    for (BlockArgument &blockArg : region.getArguments().drop_front(1)) {
+      if (!llvm::isa<IndexType>(blockArg.getType())) {
+        validArguments = false;
+        break;
+      }
+    }
+  }
+  if (!validArguments) {
+    return op->emitOpError(
+        "expected condition region to take (%device: !hal.device, "
+        "%workload_0: index, %workload_1: index, ...");
+  }
+
+  // Verify the return type is i1.
+  for (auto returnOp : region.getOps<IREE::HAL::ReturnOp>()) {
+    auto returnTypes = returnOp.getOperandTypes();
+    if (returnTypes.size() != 1 || !llvm::all_of(returnTypes, [](Type type) {
+          return type.isInteger(1);
```

Comment:
Here too if checking for signedness above
```suggestion
          return type.isSignlessInteger(1);
```